### PR TITLE
Form block layout

### PR DESCRIPTION
**Pull request by hfazai**

### CODE REVIEW

File: galite-tests/src/test/kotlin/org/kopi/galite/demo/product/ProductForm.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ import org.kopi.galite.form.dsl.ReportSelectionForm
 import org.kopi.galite.report.Report
 
 object ProductForm : ReportSelectionForm() {
-  override val locale = Locale.ENGLISH
+  override val locale = Locale.FRANCE
```
Comment (SNsir) [inline galite-tests/src/test/kotlin/org/kopi/galite/demo/product/ProductForm.kt:31]: already fixed in my PR

File: galite-tests/src/test/kotlin/org/kopi/galite/demo/client/ClientForm.kt [hunk 2]
```diff
@@ -27,8 +27,8 @@ import org.kopi.galite.report.Report
 
 object ClientForm : ReportSelectionForm() {
   override val locale = Locale.FRANCE
-  override val title = "client form"
-  val page = page("page")
+  override val title = "Clients"
```
Comment (SNsir) [inline galite-tests/src/test/kotlin/org/kopi/galite/demo/client/ClientForm.kt:30]: This class is to be fixed in my store application open PR, not here. (already fixed you can review it and merge into master)

File: galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/field/Field.kt [hunk 3]
```diff
@@ -30,11 +29,20 @@ import com.vaadin.flow.component.HasEnabled
  * @param hasDecrement has decrement button ?
  * TODO: Implement this class with appropriate component
  */
-open class Field(hasIncrement: Boolean, hasDecrement: Boolean) : Component(), HasEnabled {
+open class Field(hasIncrement: Boolean, hasDecrement: Boolean) : Div() {
```
Comment (SNsir) [inline galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/field/Field.kt:32]: Are you sure about this change?

**Review by SNsir — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/block/AbstractBlockLayout.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+   * @param rowSpan The row span width.
+   */
+  open fun setComponent(widget: Component?, column: Int, row: Int, colSpan: Int, rowSpan: Int) {
+    // TODO
+  }
+
+  open fun addAlignedComponent(widget: Component?, constraint: ComponentConstraint?) {
+    TODO()
+  }
+
```
[the remark this answers is not in the document]
Comment (hfazai) [reply galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/block/AbstractBlockLayout.kt:68]: Already exist setVisible() in `Component` class

**Review by SNsir — APPROVED** [review]

**Pull request merged**